vm/src/interpreter.cpp

```diff
@@ -121,7 +216,20 @@ void run(constant_pool pool, const char *bytes, size_t size) {
             pid_t pid = fork();
             if (pid == 0) {
                 // Replace the current process with a new process image
-                execvp(argv[0], argv);
+                if (execvp(argv[0], argv) == -1) {
+                    for (int i = 0; i < frame_size; i++) {
+                        delete[] argv[i];
+                    }
+                    delete[] argv;
```

Comment:
no need to free if you exit, and this fragment is duplicated three times

Reply:
Abusing the fact that exiting a process automatically is not worth it. Regarding the repetition for each error case, it is relatively small, but a small refactoring could effectively help here.